Mathlib/LinearAlgebra/RootSystem/Defs.lean

```diff
@@ -150,6 +150,17 @@ protected def flip : RootPairing ι R N M :=
 lemma flip_flip : P.flip.flip = P :=
   rfl
 
+/-- If we interchange the roles of `M` and `N`, we still have a root system. -/
+protected def _root_.RootSystem.flip (P : RootSystem ι R M N) : RootSystem ι R N M :=
+  { toRootPairing := P.toRootPairing.flip
+    span_root_eq_top := P.span_coroot_eq_top
+    span_coroot_eq_top := P.span_root_eq_top }
+
+@[simp]
+protected lemma _root_.RootSystem.flip_flip (P : RootSystem ι R M N) :
```

Comment:
Might be handy to state this with `Bijective` as well (`Involutive` is unfortunately not type-correct), or bundle it as `flipEquiv`

Reply:
IIUC you're suggesting a lemma stating `Bijective (RootPairing.flip (ι := ι) (R := R) (M := M) (N := N))` or similar?

I think it's not so likely this will be useful so I'd rather wait and see. No strong feelings though: I can add it if you think I really should.

Reply:
Similar remarks for my understanding of the other suggestion:
```lean
variable (ι R M N) in
def flipEquiv : RootPairing ι R N M ≃ RootPairing ι R M N where
  toFun P := P.flip
  invFun P := P.flip
  left_inv _ := rfl
  right_inv _ := rfl
```

Reply:
The application I'm thinking of is one where you'd prefer to argue by flipping all the root pairings, either via `Injective` or via `Surjective.forall`. Bijectivity follows trivially from `Equiv.bijective`, but I think it's still worth stating explicitly to get in the simp normal form.

Maybe this type of argument never comes up, but it's always handy if the construction is already in the library when a future user needs it!

Reply:
I find it very hard to imagine a situation like to one you describe arising but as a compromise, I've added `flipEquiv` for both `RootPairing` and `RootSystem`.